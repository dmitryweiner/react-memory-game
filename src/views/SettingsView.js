import React, {Component} from "react";
import Button from "@material-ui/core/Button";
import Typography from "@material-ui/core/Typography";
import Slider from "@material-ui/core/Slider";

export default class SettingsView extends Component {
  render() {
    return <>
      <Typography id="discrete-slider" gutterBottom>
        Количество чисел
      </Typography>
      <Slider
        aria-labelledby="discrete-slider"
        valueLabelDisplay="auto"
        defaultValue={3}
        step={1}
        min={1}
        max={6}
      />

      <Typography id="discrete-slider" gutterBottom>
        Величина чисел
      </Typography>
      <Slider
        aria-labelledby="discrete-slider"
        valueLabelDisplay="auto"
        defaultValue={1}
        step={1}
        min={1}
        max={5}
      />

      <Typography id="discrete-slider" gutterBottom>
        Время демонстрации
      </Typography>
      <Slider
        aria-labelledby="discrete-slider"
        valueLabelDisplay="auto"
        defaultValue={10}
        step={1}
        min={5}
        max={30}
      />

      <Button variant="contained" color="primary" onClick={() => this.props.handleTabs(1)}>Играть!</Button>
    </>;
  }
}
