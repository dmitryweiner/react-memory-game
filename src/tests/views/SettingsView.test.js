import React from "react";
import renderer from "react-test-renderer";
import SettingsView from "../../views/SettingsView";

test("MainView", () => {
  const props = {
    settings: {
      numbersCount: 3,
      numbersMagnitude: 1,
      delay: 10
    }
  };
  const component = renderer.create(<SettingsView {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
