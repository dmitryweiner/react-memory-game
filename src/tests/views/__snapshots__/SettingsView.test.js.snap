// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView 1`] = `
Array [
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    id="discrete-slider"
  >
    Количество чисел
  </p>,
  <span
    className="MuiSlider-root MuiSlider-colorPrimary"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "40%",
        }
      }
    />
    <input
      type="hidden"
      value="3"
    />
    <span
      aria-labelledby="discrete-slider"
      aria-orientation="horizontal"
      aria-valuemax={6}
      aria-valuemin={1}
      aria-valuenow={3}
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-1"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "40%",
        }
      }
      tabIndex={0}
    >
      <span
        className="PrivateValueLabel-offset-3 MuiSlider-valueLabel"
      >
        <span
          className="PrivateValueLabel-circle-4"
        >
          <span
            className="PrivateValueLabel-label-5"
          >
            3
          </span>
        </span>
      </span>
    </span>
  </span>,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    id="discrete-slider"
  >
    Величина чисел
  </p>,
  <span
    className="MuiSlider-root MuiSlider-colorPrimary"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "0%",
        }
      }
    />
    <input
      type="hidden"
      value="1"
    />
    <span
      aria-labelledby="discrete-slider"
      aria-orientation="horizontal"
      aria-valuemax={5}
      aria-valuemin={1}
      aria-valuenow={1}
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-1"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "0%",
        }
      }
      tabIndex={0}
    >
      <span
        className="PrivateValueLabel-offset-3 MuiSlider-valueLabel"
      >
        <span
          className="PrivateValueLabel-circle-4"
        >
          <span
            className="PrivateValueLabel-label-5"
          >
            1
          </span>
        </span>
      </span>
    </span>
  </span>,
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
    id="discrete-slider"
  >
    Время демонстрации
  </p>,
  <span
    className="MuiSlider-root MuiSlider-colorPrimary"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail"
    />
    <span
      className="MuiSlider-track"
      style={
        Object {
          "left": "0%",
          "width": "20%",
        }
      }
    />
    <input
      type="hidden"
      value="10"
    />
    <span
      aria-labelledby="discrete-slider"
      aria-orientation="horizontal"
      aria-valuemax={30}
      aria-valuemin={5}
      aria-valuenow={10}
      className="MuiSlider-thumb MuiSlider-thumbColorPrimary PrivateValueLabel-thumb-1"
      data-index={0}
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      role="slider"
      style={
        Object {
          "left": "20%",
        }
      }
      tabIndex={0}
    >
      <span
        className="PrivateValueLabel-offset-3 MuiSlider-valueLabel"
      >
        <span
          className="PrivateValueLabel-circle-4"
        >
          <span
            className="PrivateValueLabel-label-5"
          >
            10
          </span>
        </span>
      </span>
    </span>
  </span>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Играть!
    </span>
  </button>,
]
`;
